File: Fatorial.java
import java.util.Scanner;

public class Fatorial{
    public static void main(String[] args){
        int soma = 1;
        Scanner scan = new Scanner(System.in);
        
        System.out.println("Digite um numero para calcular o fatorial: ");
        int numero  = scan.nextInt();
        for(int index = numero; index > 0; index--){
            soma *= index;
        }
        System.out.printf("O fatorial de: %d é igual %d%n", numero, soma);
    }
}

File: Fatorial.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});

readline.question('Digite um numero para calcular o fatorial: ', value => {
  Fatorial(Number(value));
  readline.close();
});

function Fatorial(param){
    let soma = 1;
    if (typeof param !== 'number'){
        return console.log("Somente numero");
    } else {
        for(let index = param; index > 0; index--){
            soma *= index;
        }
    }
    return console.log(soma);
}